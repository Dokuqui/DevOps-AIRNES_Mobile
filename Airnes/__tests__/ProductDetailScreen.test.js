import React from "react";
import { render, waitFor } from "@testing-library/react-native";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import ProductDetailScreen from "../screens/ProductPageScreen";

jest.mock("@expo/vector-icons", () => ({
  Ionicons: "Ionicons",
}));

jest.mock("../components/util/helper.js", () => ({
  APIRequest: jest.fn((method, url) => {
    if (url.includes("Products")) {
      return Promise.resolve({
        success: true,
        return: {
          ProductId: 1,
          Name: "Product 1",
          Description: "Lorem ipsum",
          Price: "22.00",
          Stock: 10,
          Pictures: [
            {
              Link: "https://example.com/product1.jpg",
            },
          ],
        },
      });
    } else if (url.includes("ProductMaterial")) {
      return Promise.resolve({
        success: true,
        return: [
          { MaterialId: 1, Label: "Material 1" },
          { MaterialId: 2, Label: "Material 2" },
        ],
      });
    }
  }),
}));

jest.mock("../components/MainPage/Footer", () => () => <></>);

const mockStore = configureStore([]);

jest.mock("../store/basket", () => ({
  addBasketItem: jest.fn((item) => async (dispatch) => {
    return dispatch({
      type: "basket/addBasketItem/fulfilled",
      payload: item,
    });
  }),
}));

describe("ProductDetailScreen", () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      products: {
        items: [],
        status: "idle",
        error: null,
      },
    });
  });

  it("renders correctly with fetched product data", async () => {
    const { getByText, toJSON } = render(
      <Provider store={store}>
        <ProductDetailScreen
          navigation={{}}
          route={{ params: { productId: 1 } }}
        />
      </Provider>
    );

    await waitFor(() => {
      expect(getByText("Product 1")).toBeTruthy();
      expect(getByText("Lorem ipsum")).toBeTruthy();
    });

    expect(toJSON()).toMatchSnapshot();
  });
});