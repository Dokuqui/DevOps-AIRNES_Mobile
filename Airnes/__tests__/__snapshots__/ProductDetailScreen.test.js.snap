// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetailScreen renders correctly with fetched product data 1`] = `
<RCTScrollView
  style={
    {
      "marginBottom": 32,
    }
  }
>
  <View>
    <Image
      source={
        {
          "uri": "undefined/https://example.com/product1.jpg",
        }
      }
      style={
        {
          "height": 200,
          "width": "100%",
        }
      }
    />
    <Text
      style={
        {
          "fontFamily": "open-bold",
          "fontSize": 24,
          "margin": 8,
          "textAlign": "center",
        }
      }
    />
    <View
      style={
        [
          {
            "backgroundColor": "rgba(255, 255, 255, 0.7)",
            "borderColor": "#EECB98",
            "borderRadius": 8,
            "borderWidth": 1,
            "padding": 8,
          },
          {
            "marginHorizontal": "auto",
            "width": "80%",
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "column",
              "padding": 8,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "alignItems": "center",
                "fontFamily": "shinko-font",
                "fontSize": 18,
                "justifyContent": "center",
                "marginVertical": 4,
              },
              {
                "fontSize": 18,
                "marginHorizontal": "auto",
              },
            ]
          }
        >
          22.00
           â‚¬
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
        }
      }
    >
      <View
        style={
          {
            "width": "80%",
          }
        }
      >
        <View
          style={
            {
              "borderBottomColor": "#351401",
              "borderBottomWidth": 2,
              "margin": 4,
              "marginHorizontal": 12,
              "marginVertical": 4,
              "padding": 6,
            }
          }
        >
          <Text
            style={
              {
                "color": "#D89634",
                "fontFamily": "open-bold",
                "fontSize": 20,
                "textAlign": "center",
              }
            }
          >
            Title
          </Text>
        </View>
        <View
          style={
            {
              "borderRadius": 6,
              "marginHorizontal": 12,
              "marginVertical": 4,
              "paddingHorizontal": 8,
              "paddingVertical": 4,
            }
          }
        >
          <Text
            style={
              {
                "color": "#351401",
                "fontFamily": "shinko-font",
                "fontSize": 16,
                "textAlign": "center",
              }
            }
          >
            Product 1
          </Text>
        </View>
        <View
          style={
            {
              "borderBottomColor": "#351401",
              "borderBottomWidth": 2,
              "margin": 4,
              "marginHorizontal": 12,
              "marginVertical": 4,
              "padding": 6,
            }
          }
        >
          <Text
            style={
              {
                "color": "#D89634",
                "fontFamily": "open-bold",
                "fontSize": 20,
                "textAlign": "center",
              }
            }
          >
            Description
          </Text>
        </View>
        <View
          style={
            {
              "borderRadius": 6,
              "marginHorizontal": 12,
              "marginVertical": 4,
              "paddingHorizontal": 8,
              "paddingVertical": 4,
            }
          }
        >
          <Text
            style={
              {
                "color": "#351401",
                "fontFamily": "shinko-font",
                "fontSize": 16,
                "textAlign": "center",
              }
            }
          >
            Lorem ipsum
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "marginVertical": 20,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginVertical": 8,
          }
        }
      >
        <Text
          style={
            {
              "fontFamily": "open-bold",
              "marginRight": 8,
            }
          }
        >
          Quantity:
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "white",
              "borderRadius": 25,
              "marginHorizontal": 5,
              "opacity": 1,
              "paddingHorizontal": 20,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "black",
                "fontFamily": "open-bold",
                "fontSize": 16,
                "textAlign": "center",
              }
            }
          >
            -
          </Text>
        </View>
        <Text
          style={
            {
              "fontFamily": "open-bold",
              "fontSize": 16,
              "marginHorizontal": 8,
            }
          }
        >
          1
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "white",
              "borderRadius": 25,
              "marginHorizontal": 5,
              "opacity": 1,
              "paddingHorizontal": 20,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "black",
                "fontFamily": "open-bold",
                "fontSize": 16,
                "textAlign": "center",
              }
            }
          >
            +
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginVertical": 8,
          }
        }
      >
        <Text
          style={
            {
              "fontFamily": "open-bold",
              "marginRight": 8,
            }
          }
        >
          Color:
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "red",
              "borderRadius": 12,
              "height": 24,
              "marginHorizontal": 4,
              "opacity": 1,
              "width": 24,
            }
          }
          testID="color-selector-red"
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "blue",
              "borderRadius": 12,
              "height": 24,
              "marginHorizontal": 4,
              "opacity": 1,
              "width": 24,
            }
          }
          testID="color-selector-blue"
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "green",
              "borderRadius": 12,
              "height": 24,
              "marginHorizontal": 4,
              "opacity": 1,
              "width": 24,
            }
          }
          testID="color-selector-green"
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "marginVertical": 8,
          }
        }
      >
        <Text
          style={
            {
              "fontFamily": "shinko-font",
            }
          }
        >
          Available:
           
          <Text
            style={
              [
                {
                  "fontFamily": "open-bold",
                },
                {
                  "color": "#00C517",
                },
              ]
            }
          >
            In Stock
          </Text>
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": "auto",
            "marginVertical": 20,
            "width": 300,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#F4A460",
              "borderRadius": 15,
              "marginVertical": 8,
              "opacity": 1,
              "paddingHorizontal": 15,
              "paddingVertical": 8,
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "fontFamily": "open-bold",
                "fontSize": 14,
              }
            }
          >
            Add to Basket
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
